rai/node/wallet.cpp

```diff
@@ -1186,116 +1186,41 @@ void rai::wallet::work_ensure (MDB_txn * transaction_a, rai::account const & acc
+					if (node.config.receive_minimum.number () <= amount)
 					{
+						BOOST_LOG (node.log) << boost::str (boost::format ("Found a pending block %1% for account %2%") % hash.to_string () % pending.source.to_account ());
 						auto this_l (shared_from_this ());
-						std::shared_ptr<rai::block> block_l (wallet->node.store.block_get (transaction, info.head));
-						wallet->node.background ([this_l, account, block_l] {
-							this_l->wallet->node.active.start (block_l, [this_l, account](std::shared_ptr<rai::block>) {
-								// If there were any forks for this account they've been rolled back and we can receive anything remaining from this account
-								this_l->receive_all (account);
-							});
-							this_l->wallet->node.network.broadcast_confirm_req (block_l);
-						});
-						already_searched.insert (account);
-					}
-				}
-				else
-				{
-					BOOST_LOG (wallet->node.log) << boost::str (boost::format ("Not receiving block %1% due to minimum receive threshold") % key.hash.to_string ());
-				}
-			}
-		}
-		BOOST_LOG (wallet->node.log) << "Pending block search phase complete";
-	}
-	void receive_all (rai::account const & account_a)
-	{
-		BOOST_LOG (wallet->node.log) << boost::str (boost::format ("Account %1% confirmed, receiving all blocks") % account_a.to_account ());
-		rai::transaction transaction (wallet->node.store.environment, nullptr, false);
-		auto representative (wallet->store.representative (transaction));
-		for (auto i (wallet->node.store.pending_begin (transaction)), n (wallet->node.store.pending_end ()); i != n; ++i)
-		{
-			rai::pending_key key (i->first);
-			rai::pending_info pending (i->second);
-			if (pending.source == account_a)
-			{
-				if (wallet->store.exists (transaction, key.account))
-				{
-					if (wallet->store.valid_password (transaction))
-					{
-						rai::pending_key key (i->first);
-						std::shared_ptr<rai::block> block (wallet->node.store.block_get (transaction, key.hash));
-						auto wallet_l (wallet);
-						auto amount (pending.amount.number ());
-						BOOST_LOG (wallet_l->node.log) << boost::str (boost::format ("Receiving block: %1%") % block->hash ().to_string ());
-						wallet_l->receive_async (block, representative, amount, [wallet_l, block](std::shared_ptr<rai::block> block_a) {
-							if (block_a == nullptr)
-							{
-								BOOST_LOG (wallet_l->node.log) << boost::str (boost::format ("Error receiving block %1%") % block->hash ().to_string ());
-							}
-						},
-						true);
-					}
-					else
-					{
-						BOOST_LOG (wallet->node.log) << boost::str (boost::format ("Unable to fetch key for: %1%, stopping pending search") % key.account.to_account ());
+						rai::account_info info;
+						auto error (node.store.account_get (transaction, pending.source, info));
+						assert (!error);
+						std::shared_ptr<rai::block> block_l (node.store.block_get (transaction, info.head));
+						node.active.start (block_l);
+						node.network.broadcast_confirm_req (block_l);
```

Comment:
I think it's better to call `node.block_confirm (block_l)` in replacement of active.start + broadcast_confirm_req.